social_core/tests/backends/test_orcid.py

```diff
@@ -0,0 +1,26 @@
+import json
+
+from .oauth import OAuth2Test
+
+from social_core.backends.orcid import ORCIDOAuth2
+
+class ORCIDOAuth2Test(OAuth2Test):
+    backend_path = 'social_core.backends.orcid.ORCIDOAuth2'
+    user_data_url = 'https://orcid.org/oauth/userinfo'
+    expected_username = '0000-0002-2601-8132'
+    access_token_body = json.dumps({
+        'access_token': 'foobar',
+        'token_type': 'bearer'
```

Comment:
```suggestion
        'token_type': 'bearer',
        'orcid-identifier': {'path': '0000-0002-2601-8132'},
```

This is how you extract username in the backend, so you should inject it here as well (or at least I hope I understand your code). Anyway put here the expected OAuth response which is parsed `by get_user_details`.